logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java

```diff
@@ -72,13 +73,23 @@ public void replicateLatestSnapshot(final Consumer<Runnable> executor) {
       LOG.debug("Start replicating latest snapshot {}", latestSnapshotDirectory);
 
       try (final var stream = Files.list(latestSnapshotDirectory)) {
-        final var files = stream.collect(Collectors.toList());
-        for (final var file : files) {
+        final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+        final var paths = stream.sorted().collect(Collectors.toList());
+        for (final var path : paths) {
+          outStream.write(Files.readAllBytes(path));
```

Comment:
This can be problematic. Snapshot can be too large to fit in memory.

Reply:
One way to calculate checksums for a directory is to calculate checksum for each file, then calculate checksum of the checksums.